crates/uv-resolver/src/lock.rs

```diff
@@ -596,85 +592,98 @@ pub(crate) struct SourceDist {
 }
 
 impl SourceDist {
-    fn from_resolved_dist(resolved_dist: &ResolvedDist) -> Result<Option<SourceDist>, LockError> {
-        match *resolved_dist {
+    fn from_annotated_dist(
+        annotated_dist: &AnnotatedDist,
+    ) -> Result<Option<SourceDist>, LockError> {
+        match annotated_dist.dist {
             // TODO: Do we want to try to lock already-installed distributions?
             // Or should we return an error?
             ResolvedDist::Installed(_) => todo!(),
-            ResolvedDist::Installable(ref dist) => SourceDist::from_dist(dist),
+            ResolvedDist::Installable(ref dist) => {
+                SourceDist::from_dist(dist, &annotated_dist.hashes)
+            }
         }
     }
 
-    fn from_dist(dist: &Dist) -> Result<Option<SourceDist>, LockError> {
+    fn from_dist(dist: &Dist, hashes: &[HashDigest]) -> Result<Option<SourceDist>, LockError> {
         match *dist {
             Dist::Built(_) => Ok(None),
-            Dist::Source(ref source_dist) => SourceDist::from_source_dist(source_dist).map(Some),
+            Dist::Source(ref source_dist) => {
+                SourceDist::from_source_dist(source_dist, hashes).map(Some)
+            }
         }
     }
 
     fn from_source_dist(
         source_dist: &distribution_types::SourceDist,
+        hashes: &[HashDigest],
```

Comment:
Don't love making `hashes` an argument everywhere but seemed like the most straightforward option (since it only exists on `AnnotatedDist`).

Reply:
Yeah hmmm, I don't like it either.

I think when I was writing this, I was assuming the hashes would be added to types like `PathSourceDist` directly. Out of curiosity, why didn't you take that approach? AIUI, hashes are on _some_ of the distribution types but not all of them. I was thinking we would make that a bit more consistent.

Reply:
They're not quite the same. The hash on the registry distribution is just the hash reported by the registry, _not_ the hash we compute locally. And the distribution exists well before we compute its hashes.

Reply:
We don't compute hashes for registry distributions during resolution. We only _validate_ them during installation.